native/c/zdsm.c

```diff
@@ -85,3 +90,125 @@ typedef int (*IGWASMS)(
     ZSMS_DATA sms_data[3],
     int *ds_type // 1 for PDSE, 2 for HFS
     ) ATTRIBUTE(amode31);
+
+#pragma prolog(ZDSRECFM, " ZWEPROLG NEWDSA=(YES,128) ")
+#pragma epilog(ZDSRECFM, " ZWEEPILG ")
+
+// Obtain the record format for a data set, given its name and volser
+// Full PDF for advanced services: https://www.ibm.com/docs/en/SSLTBW_2.2.0/pdf/dgt3s310.pdf
+// Doc page: https://www.ibm.com/docs/en/zos/3.1.0?topic=macros-reading-dscbs-from-vtoc-using-obtain
+int ZDSRECFM(ZDS *zds, const char *dsn, const char *volser, char *recfm_buf,
+             int recfm_buf_len)
+{
+  struct DSCBFormat1 *dscb = NULL;
+  // workarea: each DSCB is 140 bytes, we need enough space for format-1 DSCB, format-8 DSCB and max possible format-3 DSCBs (10)
+  // adding 140 bytes for the workarea itself
+  char workarea[0x8C * (MAX_DSCBS + 1)];
+  struct ObtainParams params;
```

Comment:
`struct` is not needed here because of typedef

Reply:
good point, fixed in upcoming commit 👍